File: data structures and algorithms/java/04_003.java
class Rectangle {
    protected int width;
    protected int height;
 
    public void setWidth(int width) {
        this.width = width;
    }
 
    public void setHeight(int height) {
        this.height = height;
    }
 
    public int getArea() {
        return width * height;
    }
}
 
class Square extends Rectangle {
    @Override
    public void setWidth(int width) {
        this.width = width;
        this.height = width;
    }
 
    @Override
    public void setHeight(int height) {
        this.height = height;
        this.width = height;
    }
}
 
public class LSPExample {
    public static void main(String[] args) {
        Rectangle rectangle = new Rectangle();
        rectangle.setWidth(5);
        rectangle.setHeight(10);
        int rectangleArea = rectangle.getArea();
        System.out.println("Area of rectangle: " + rectangleArea);
 
        Square square = new Square();
        square.setWidth(5);
        int squareArea = square.getArea();
        System.out.println("Area of square: " + squareArea);
    }
}

File: data structures and algorithms/cs/04_003.cs
class Rectangle {
    protected int width;
    protected int height;

    public void SetWidth(int width) {
        this.width = width;
    }

    public void SetHeight(int height) {
        this.height = height;
    }

    public int GetArea() {
        return width * height;
    }
}

class Square : Rectangle {
    public new void SetWidth(int width) {
        this.width = width;
        this.height = width;
    }

    public new void SetHeight(int height) {
        this.height = height;
        this.width = height;
    }
}

public class LSPExample {
    public static void Main(string[] args) {
        Rectangle rectangle = new Rectangle();
        rectangle.SetWidth(5);
        rectangle.SetHeight(10);
        int rectangleArea = rectangle.GetArea();
        Console.WriteLine("Area of rectangle: " + rectangleArea);

        Square square = new Square();
        square.SetWidth(5);
        int squareArea = square.GetArea();
        Console.WriteLine("Area of square: " + squareArea);
    }
}
